tests/transformers/test_numerical.py

```diff
@@ -33,6 +33,223 @@ def test_fit(self):
+        assert transformer.rounding == expected_digits
+        assert transformer.rounding_digits == expected_digits
+
+    def test_fit_rounding_auto(self):
+        """Test fit rounding parameter with ``'auto'``
+
+        If the ``rounding`` parameter is set to ``'auto'``,
+        fit should learn the ``rounding_digits`` to be the max
+        number of decimal places seen in the data.
+        """
+        # Setup
+        data = np.array([1, 2.1, 3.12, 4.123, 5.1234, 6.123, 7.12, 8.1, 9])
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan='nan', rounding='auto')
+        transformer.fit(data)
+
+        # Asserts
+        assert transformer.rounding_digits == 4
+
+    def test_fit_rounding_auto_large_numbers(self):
+        """Test fit rounding parameter with ``'auto'``
+
+        If the ``rounding`` parameter is set to ``'auto'``
+        and the data is very large, fit should learn
+        ``rounding_digits`` to be the biggest number of 0s
+        to round to that keeps the data the same.
+
+        Input:
+        - Array of data with numbers as big as 10^20
+        """
+        # Setup
+        exponents = [np.random.randint(10, 20) for i in range(10)]
+        big_numbers = [10**exponents[i] for i in range(10)]
+        data = np.array(big_numbers)
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan='nan', rounding='auto')
+        transformer.fit(data)
+
+        # Asserts
+        expected = -min(exponents) if min(exponents) > 0 else None
+        assert transformer.rounding_digits == expected
+
+    def test_fit_rounding_auto_max_decimals(self):
+        """Test fit rounding parameter with ``'auto'``
+
+        If the ``rounding`` parameter is set to ``'auto'``,
+        fit should learn the ``rounding_digits`` to be the max
+        number of decimal places seen in the data. The biggest
+        amount of decimals that can be accurately compared with
+        floats is 15.
+
+        Input:
+        - Array with a value that has over 15 decimals.
+        """
+        # Setup
+        data = np.array([0.000000000000001])
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan='nan', rounding='auto')
+        transformer.fit(data)
+
+        # Asserts
+        assert transformer.rounding_digits is None
+
+    def test_reverse_transform_rounding_none(self):
+        """Test ``reverse_transform`` when ``rounding`` is ``None``
+
+        The data should not be rounded at all.
+        """
+        # Setup
+        fitted_data = np.array([1, 2.1, 3.12, 4.123, 5.1234, 6.123, 7.12, 8.1, 9])
+        data = np.random.random(10)
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan='nan', rounding=None)
+        transformer.fit(fitted_data)
+        result = transformer.reverse_transform(data)
+
+        # Assert
+        assert (result == data).all()
+
+    def test_reverse_transform_rounding_auto_decimals(self):
```

Comment:
I don't think `>15` is actually a special case. It only matters when we are trying to determine the digits to round to since we can't accurately be more granular than that. But I think users should be able to provide that if they want, and it will behave the same as any other positive number except it might not always be accurate